pkg/apis/management.cattle.io/v3/pod_security_types.go

```diff
@@ -0,0 +1,92 @@
+package v3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const PodSecurityAdmissionConfigurationTemplateKind = "PodSecurityAdmissionConfigurationTemplate"
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type PodSecurityAdmissionConfigurationTemplate struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Configuration PodSecurityAdmissionConfigurationTemplateSpec `json:"configuration"`
+}
+
+type PodSecurityAdmissionConfigurationTemplateSpec struct {
+	Defaults   PodSecurityAdmissionConfigurationTemplateDefaults   `json:"defaults"`
+	Exemptions PodSecurityAdmissionConfigurationTemplateExemptions `json:"exemptions,omitempty"`
+}
+
+// PodSecurityAdmissionConfigurationTemplateDefaults are applied when a mode label is not set.
+//
+// The level label values must be one of
+// 'privileged' (default), 'baseline', or 'restricted'
+//
+// The version levels must be either 'latest' (default), or a specific version (e.g. 'v1.25')
+type PodSecurityAdmissionConfigurationTemplateDefaults struct {
+	Enforce        string `json:"enforce"`
+	EnforceVersion string `json:"enforce-version" yaml:"enforce-version"`
```

Comment:
It's somewhat baffling that there are different naming conventions here, like `enforce-version` vs. `runtimeClasses` but that's how it's in upstream. So that's OK we follow it.